src/bson.rs

```diff
@@ -801,23 +770,29 @@ impl Bson {
     }
 }
 
-/// `TimeStamp` representation in struct for serde serialization
-///
-/// Just a helper for convenience
-///
-/// ```rust,ignore
-/// use serde::{Serialize, Deserialize};
-/// use bson::TimeStamp;
-///
-/// #[derive(Serialize, Deserialize)]
-/// struct Foo {
-///     timestamp: TimeStamp,
-/// }
-/// ```
+/// Represents a BSON timestamp value.
 #[derive(Debug, Eq, PartialEq, Clone, Copy, Hash)]
 pub struct TimeStamp {
-    pub t: u32,
-    pub i: u32,
+    pub time: u32,
+    pub increment: u32,
+}
+
+impl TimeStamp {
+    pub(crate) fn to_le_i64(self) -> i64 {
+        let upper = (self.time.to_le() as u64) << 32;
+        let lower = self.increment.to_le() as u64;
+
+        (upper | lower) as i64
+    }
+
+    pub(crate) fn from_le_i64(val: i64) -> Self {
+        let ts = val.to_le();
+
+        TimeStamp {
+            time: ((ts as u64) >> 32) as u32,
```

Comment:
This arithmetic is moved from [here](https://github.com/mongodb/bson-rust/blob/9055e1833f872eed6e5f5e116512e6d401787964/src/decoder/serde.rs#L682-L685)